homeassistant/components/climate/ecobee.py

```diff
@@ -246,25 +264,38 @@ def turn_away_mode_off(self):
 
     def set_temperature(self, **kwargs):
         """Set new target temperature."""
-        if kwargs.get(ATTR_TARGET_TEMP_LOW) is not None and \
-           kwargs.get(ATTR_TARGET_TEMP_HIGH) is not None:
-            high_temp = int(kwargs.get(ATTR_TARGET_TEMP_LOW))
-            low_temp = int(kwargs.get(ATTR_TARGET_TEMP_HIGH))
+        low_temp = None
+        high_temp = None
+        temp = None
+
+        if kwargs.get(ATTR_TARGET_TEMP_LOW) is not None:
+            low_temp = int(kwargs.get(ATTR_TARGET_TEMP_LOW))
+
+        if kwargs.get(ATTR_TARGET_TEMP_HIGH) is not None:
+            high_temp = int(kwargs.get(ATTR_TARGET_TEMP_HIGH))
+
+        if kwargs.get("temperature") is not None:
+            temp = int(kwargs.get("temperature"))
+
+        if self.current_operation == STATE_HEAT:
+            low_temp = temp
+            high_temp = temp + 20
+        elif self.current_operation == STATE_COOL:
+            low_temp = temp - 20
+            high_temp = temp
```

Comment:
We have to check that `low_temp` and `high_temp` is not None before proceeding and calling the ecobee python API. I'd simplify somewhat like this:

```py
def set_temperature(self, **kwargs):
    """Set new target temperature."""
    low_temp = kwargs.get(ATTR_TARGET_TEMP_LOW)
    high_temp = kwargs.get(ATTR_TARGET_TEMP_HIGH)
    temp = kwargs.get(ATTR_TEMPERATURE)

    if self.current_operation == STATE_HEAT and temp is not None:
        low_temp = temp
        high_temp = temp + 20
    elif self.current_operation == STATE_COOL and temp is not None:
        low_temp = temp - 20
        high_temp = temp
    if low_temp is None and high_temp is None:
        _LOGGER.error(
            'Missing valid arguments for set_temperature in %s', kwargs)
        return
    if self.hold_temp:
        self.data.ecobee.set_hold_temp(
            self.thermostat_index, int(high_temp), int(low_temp), "indefinite")
    else:
        self.data.ecobee.set_hold_temp(
            self.thermostat_index, int(high_temp), int(low_temp))
    _LOGGER.debug(...)
```